tests/TestCase/View/Helper/UrlHelperTest.php

```diff
@@ -544,4 +544,35 @@ public function testWebrootPaths()
 
         Configure::write('App.wwwRoot', $webRoot);
     }
+
+    /**
+     * Test plugin based assets will NOT use the plugin name
+     *
+     * @return void
+     */
+    public function testPluginAssetsPrependImageBaseUrl()
+    {
+        $cdnPrefix = 'https://cdn.example.com/';
+        $imageBaseUrl = Configure::read('App.imageBaseUrl');
+        $jsBaseUrl = Configure::read('App.jsBaseUrl');
+        $cssBaseUrl = Configure::read('App.cssBaseUrl');
+        Configure::write('App.imageBaseUrl', $cdnPrefix);
+        $result = $this->Helper->image('TestTheme.text.jpg');
+        $expected = $cdnPrefix . 'text.jpg';
+        $this->assertSame($expected, $result);
+
+        Configure::write('App.jsBaseUrl', $cdnPrefix);
+        $result = $this->Helper->script('TestTheme.app.js');
+        $expected = $cdnPrefix . 'app.js';
+        $this->assertSame($expected, $result);
+
+        Configure::write('App.cssBaseUrl', $cdnPrefix);
+        $result = $this->Helper->css('TestTheme.app.css');
+        $expected = $cdnPrefix . 'app.css';
+        $this->assertSame($expected, $result);
+
+        Configure::write('App.imageBaseUrl', $imageBaseUrl);
+        Configure::write('App.jsBaseUrl', $jsBaseUrl);
+        Configure::write('App.cssBaseUrl', $cssBaseUrl);
```

Comment:
You shouldn't need to manually restore these values. The test case base class snapshots configure on each test.

Reply:
fixed, a little too paranoid...